streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -486,6 +468,23 @@ public boolean conditionMet() {
         assertEquals(thread.state(), StreamThread.State.DEAD);
     }
 
+    private StreamThread createStreamThread(final String clientId, final StreamsConfig config, final boolean eosEnabled) {
+        if (eosEnabled) {
+            clientSupplier = new MockClientSupplier(applicationId);
+        }
+        return StreamThread.create(internalTopologyBuilder,
```

Comment:
nit: in unit test we may want to avoid calling `create` because it will construct a lot of unnecessary modules like task manager / created / clients etc, we should consider mocking all these in `StreamThreadTest`.